core/storage/improvements/gae_models.py

```diff
@@ -0,0 +1,257 @@
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models related to Oppia improvement tasks."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+import feconf
+
+from google.appengine.ext import ndb
+
+(base_models,) = models.Registry.import_models([models.NAMES.base_model])
+
+ENTITY_TYPE_EXPLORATION = feconf.ENTITY_TYPE_EXPLORATION
+ENTITY_TYPES = (
+    ENTITY_TYPE_EXPLORATION,
+)
+
+STATUS_OPEN = feconf.TASK_STATUS_OPEN
+STATUS_OBSOLETE = feconf.TASK_STATUS_OBSOLETE
+STATUS_RESOLVED = feconf.TASK_STATUS_RESOLVED
+STATUS_CHOICES = (
+    STATUS_OPEN,
+    STATUS_OBSOLETE,
+    STATUS_RESOLVED,
+)
+
+TARGET_TYPE_STATE = feconf.TASK_TARGET_TYPE_STATE
+TARGET_TYPES = (
+    TARGET_TYPE_STATE,
+)
+
+TASK_TYPE_HIGH_BOUNCE_RATE = feconf.TASK_TYPE_HIGH_BOUNCE_RATE
+TASK_TYPE_INEFFECTIVE_FEEDBACK_LOOP = feconf.TASK_TYPE_INEFFECTIVE_FEEDBACK_LOOP
+TASK_TYPE_SUCCESSIVE_INCORRECT_ANSWERS = (
+    feconf.TASK_TYPE_SUCCESSIVE_INCORRECT_ANSWERS)
+TASK_TYPE_NEEDS_GUIDING_RESPONSES = feconf.TASK_TYPE_NEEDS_GUIDING_RESPONSES
+TASK_TYPES = (
+    TASK_TYPE_HIGH_BOUNCE_RATE,
+    TASK_TYPE_INEFFECTIVE_FEEDBACK_LOOP,
+    TASK_TYPE_SUCCESSIVE_INCORRECT_ANSWERS,
+    TASK_TYPE_NEEDS_GUIDING_RESPONSES,
+)
+
+
+class TaskEntryModel(base_models.BaseModel):
+    """Task entry corresponding to an actionable task in the improvements tab.
+
+    Instances have an ID of the form:
+        [entity_type].[entity_id].[entity_version].[task_type].[target_type].
+        [target_id]
+    """
+    # The type of entity a task entry refers to.
+    entity_type = ndb.StringProperty(
+        required=True, indexed=True, choices=ENTITY_TYPES)
+    # The ID of the entity a task entry refers to.
+    entity_id = ndb.StringProperty(
+        required=True, indexed=True)
+    # The version of the entity a task entry refers to.
+    entity_version = ndb.IntegerProperty(
+        required=True, indexed=True)
+    # The type of task a task entry tracks.
+    task_type = ndb.StringProperty(
+        required=True, indexed=True, choices=TASK_TYPES)
+    # The type of sub-entity a task entry focuses on. Value is None when an
+    # entity does not have any meaningful sub-entities to target.
+    target_type = ndb.StringProperty(
+        required=True, indexed=True, choices=TARGET_TYPES)
+    # Uniquely identifies the sub-entity a task entry focuses on. Value is None
+    # when an entity does not have any meaningful sub-entities to target.
+    target_id = ndb.StringProperty(
+        required=True, indexed=True)
+
+    # Utility field for performing quick look-ups by a specific entity.
```

Comment:
Can you add a comment here explaining that "This gives a speedup of X% over querying by the individual fields."?

Reply:
Done.